### PR TITLE
HPCC-16613 DESDL to allow freehand configuration

### PR DESCRIPTION
**Pull request by miguelvazq**
Fixes: 16613 + 16744

Signed-off by: Miguel Vazquez <miguel.vazquez@lexisnexis.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [x] My change requires a change to the documentation.
  - [x] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [x] This is a user interface / front-end modification
  - [x] I have tested my changes in multiple modern browers
  - [x] The component(s) render as expected

### CODE REVIEW

**Comment by miguelvazq** [comment]
@GordonSmith Please review
@rpastrana Please review translations
@kunalaswani Please confirm testing in FF
@g-pan FYI for docs https://track.hpccsystems.com/browse/HPCC-17764

**Comment by kunalaswani** [comment]
Tested on FF and working, autosave, save, refresh, all fields, tabs. All functioning as expected

**Comment by GordonSmith** [comment]
@miguelvazq build failing on:
```
/home/ati/smoketest/PR-10083/build/esp/src/build/eclwatch/DynamicESDLMethodWidget.js:
DynamicESDLMethodWidget.js.uncompressed.js:87: ERROR - Parse error. IE8 (and below) will parse trailing commas in array and object literals incorrectly. If you are targeting newer versions of JS, set the appropriate language_in option.
                onClick: function (evt) {
                ^

1 error(s), 0 warning(s)
Done (compile time:0.007s)
```

**Comment by AttilaVamos** [comment]
The ECLWatch build error is:
```
/home/ati/smoketest/PR-10083/build/esp/src/build/eclwatch/DynamicESDLMethodWidget.js:
DynamicESDLMethodWidget.js.uncompressed.js:87: ERROR - Parse error. IE8 (and below) will parse trailing commas in array and object literals incorrectly. If you are targeting newer versions of JS, set the appropriate language_in option.
                onClick: function (evt) {
                ^

1 error(s), 0 warning(s)
Done (compile time:0.007s)
```

**Comment by miguelvazq** [comment]
@GordonSmith Updated should pass now. Let's wait and see.

File: esp/src/eclwatch/nls/es/hpcc.js [hunk 1]
```diff
@@ -338,6 +338,7 @@ define(
     MaxSize: "Tamaňo máximo",
     MemberOf: "Miembro de",
     Members: "Miembros",
+    MethodConfiguration: "Configuracion de método",
```
Comment (rpastrana) [inline esp/src/eclwatch/nls/es/hpcc.js]: We capitalized both words in English, should we capitalize Método?

File: esp/src/eclwatch/nls/es/hpcc.js [hunk 2]
```diff
@@ -594,6 +595,7 @@ define(
     Subgraphs: "Sub-gráficos",
     Submit: "Enviar",
     Subtype: "Subtipo",
+    SuccessfullySaved: "Guardado correctamente",
```
Comment (rpastrana) [inline esp/src/eclwatch/nls/es/hpcc.js]: For aesthetics, I'd prob shorten this to "Guardado"

**Review by rpastrana — COMMENTED** [review]
@miguelvazq a couple of minor questions/comments.

**Comment by AttilaVamos** [comment]
The regression test errors are not related to this PR.

File: esp/src/eclwatch/DynamicESDLMethodWidget.js [hunk 3]
```diff
@@ -130,195 +83,81 @@ define([
...
-                        dynamicAttributes += " " + row.Attribute + "=\'" + row.Value +"\'";
-                        dynamicAttributes += " " + attr + "=\'" + val + "\'";
-                    });
-                    passInXML = "<Methods><Method name='" + row.Name + "\'" + dynamicAttributes + "/></Methods>"
-                });
-            }  else {
-                var passInXML = "<Methods><Method name='" + selection[0].__hpcc_id + "' " + attr + "='" + val +  "\'" + "/></Methods>";
-            }
-            WsESDLConfig.ConfigureESDLBindingMethod({
+            var xmlBuilder = "<Methods>" + userXML +  "</Methods>";
```
Comment (GordonSmith) [inline esp/src/eclwatch/DynamicESDLMethodWidget.js]: We could validate this is valid XML (in the future).

**Review by GordonSmith — APPROVED** [review]
Looks good.

**Comment by richardkchapman** [comment]
@rpastrana You happy now?

**Comment by rpastrana** [comment]
@miguelvazq spanish translations look fine.

**Comment by miguelvazq** [comment]
@richardkchapman Ready for merge.

**Comment by miguelvazq** [comment]
FYI @g-pan
